apps/studio/components/layouts/TableEditorLayout/ExportAllRows.tsx

```diff
@@ -0,0 +1,431 @@
+        }
+    ))
+
+type UseExportAllRowsReturn = {
+  exportInDesiredFormat: () => Promise<void>
+  confirmationModal: ReactNode | null
+}
+
+export const useExportAllRowsGeneric = (
+  params: UseExportAllRowsParams & OutputCallbacks
+): UseExportAllRowsReturn => {
+  const queryClient = useQueryClient()
+  const {
+    startProgressTracker,
+    trackPercentageProgress,
+    stopTrackerWithError,
+    dismissTrackerSilently,
+    markTrackerComplete,
+  } = useProgressToasts()
+
+  const { convertToOutputFormat, convertToBlob, save } = params
+
+  const [confirmationMessage, setConfirmationMessage] = useState<string | null>(null)
+
+  const exportInternal = useStaticEffectEvent(
+    async ({ bypassConfirmation }: { bypassConfirmation: boolean }): Promise<void> => {
+      if (!params.enabled) return
+
+      const { projectRef, connectionString, entity, totalRows } = params
+
+      startProgressTracker({
+        id: entity.id,
+        name: entity.name,
+        trackPercentage: totalRows !== undefined,
+      })
+
+      const exportResult =
+        params.type === 'provided_rows'
+          ? convertAndDownload(formatRowsForExport(params.rows, params.table), params.table, {
+              convertToOutputFormat,
+              convertToBlob,
+              save,
+            })
+          : await fetchAllRows({
+              queryClient,
+              projectRef: projectRef,
+              connectionString: connectionString,
+              entity: entity,
+              bypassConfirmation,
+              filters: params.filters,
+              sorts: params.sorts,
+              roleImpersonationState: params.roleImpersonationState,
+              progressCallback: totalRows
+                ? (value: number) =>
+                    trackPercentageProgress({
+                      id: entity.id,
+                      name: entity.name,
+                      totalRows: totalRows,
+                      value,
+                    })
+                : undefined,
+              convertToOutputFormat,
+              convertToBlob,
+              save,
+            })
+
+      if (exportResult.status === 'error') {
+        const error = exportResult.error
+        if (error instanceof NoRowsToExportError) {
+          return stopTrackerWithError(
+            entity.id,
+            entity.name,
+            `The table ${entity.name} has no rows to export.`
+          )
+        }
+        if (error instanceof TableTooLargeError) {
+          return stopTrackerWithError(entity.id, entity.name, MAX_EXPORT_ROW_COUNT_MESSAGE)
+        }
+        console.error(
+          `Export All Rows > Error: %s%s%s`,
+          error.message,
+          error.cause?.message ? `\n${error.cause.message}` : '',
+          error.cause?.stack ? `:\n${error.cause.stack}` : ''
+        )
+        return stopTrackerWithError(entity.id, entity.name)
+      }
+
+      if (exportResult.status === 'require_confirmation') {
```

Comment:
confirmation modal shows up with progress toast in this case - reckon we only show the toast after confirming? I'm guessing its the same with the other statuses as well since `startTrackerProgress` is being called prior to the status checks

<img width="988" height="667" alt="image" src="https://github.com/user-attachments/assets/391022f3-d2ad-4303-93da-2c09651b83df" />